### PR TITLE
HOSTEDCP-1229: Move azure cloud provider to out of tree

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Updates the cloud manager to reconcile Azure

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1229](https://issues.redhat.com/browse/HOSTEDCP-1229)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go [hunk 1]
```diff
@@ -2114,3 +2120,134 @@ func allLoadBalancersRemoved(ctx context.Context, c client.Client) (bool, error)
...
+		clusterRoleBinding.RoleRef = rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     clusterRole.Name,
+		}
+		clusterRoleBinding.Subjects = []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Namespace: "kube-system",
+				Name:      clusterRole.Name,
```
Comment (muraee) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:2170]:
```suggestion
				Name:      serviceAccount.Name,
```

Comment (muraee) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:2170]:
```suggestion
				Namespace: serviceAccount.Namespace,
				Name:      serviceAccount.Name,
```

File: control-plane-operator/controllers/hostedcontrolplane/cloud/azure/reconcile.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+func ccmLabels() map[string]string {
+	return map[string]string{
+		"app": "cloud-controller-manager",
+	}
+}
+
+func additionalLabels() map[string]string {
+	return map[string]string{
+		hyperv1.ControlPlaneComponent:       "cloud-controller-manager",
+		config.NeedManagementKASAccessLabel: "true",
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/cloud/azure/reconcile.go:120]: please drop `NeedManagementKASAccessLabel` label

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/cloudcontrollermanager/azure/cloud_node_manager.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package azure
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: implement https://github.com/kubernetes-sigs/cloud-provider-azure/blob/e5d670328a51e31787fc949ddf41a3efcd90d651/examples/out-of-tree/cloud-node-manager.yaml
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/cloudcontrollermanager/azure/cloud_node_manager.go:10]: please remove this

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go [hunk 4]
```diff
@@ -2114,3 +2120,134 @@ func allLoadBalancersRemoved(ctx context.Context, c client.Client) (bool, error)
...
+	serviceAccount := ccm.CloudNodeManagerServiceAccount()
+	if _, err := r.CreateOrUpdate(ctx, r.client, serviceAccount, func() error { return nil }); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile %T %s: %w", serviceAccount, serviceAccount.Name, err))
+	}
+
+	clusterRole := ccm.CloudNodeManagerClusterRole()
+	if _, err := r.CreateOrUpdate(ctx, r.client, clusterRole, func() error {
+		clusterRole.Rules = []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:2136]: please add a TODO to explore scoping down rbac to the running Node

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go [hunk 5]
```diff
@@ -2114,3 +2120,134 @@ func allLoadBalancersRemoved(ctx context.Context, c client.Client) (bool, error)
...
+									corev1.ResourceMemory: resource.MustParse("512Mi"),
+									corev1.ResourceCPU:    resource.MustParse("2000m"),
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		// TODO (alberto): apply this
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:2245]: Please drop this todo

**Comment by enxebre** [comment]
/retitle "Move azure cloud provider to out of tree"

**Comment by enxebre** [comment]
Thanks! After addressing comments can you please squash into one single commit?

**Comment by bryan-cox** [comment]
/retitle HOSTEDCP-1229: Move azure cloud provider to out of tree

**Comment by enxebre** [comment]
Linking for reference
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/e5d670328a51e31787fc949ddf41a3efcd90d651/examples/out-of-tree/cloud-controller-manager.yaml#L179

https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/release-4.15/pkg/cloud/azure/assets/cloud-node-manager-daemonset.yaml

**Pull request merged**